cbc-repeated-iv/src/cbc_game.rs

```diff
@@ -0,0 +1,397 @@
+
+".to_string()
+        }
+    }
+
+    pub fn start(&mut self) -> anyhow::Result<()> {
+        let mut iter = 0;
+
+        println!("{}", self.cbc_same_iv_ascii_art.on_color(Color::Green));
+
+        self.reinitialize_enc_dec();
+
+        self.encrypt_all_texts();
+
+        self.print_known_texts();
+
+        while iter < self.ciphertexts_vec.len() {
+            let mut line = String::new();
+            println!(
+                "\n{}",
+                "❱".repeat(40).color(Color::Black).on_color(Color::Yellow)
+            );
+            println!("Input key (c(continue)/q(quit)):");
+            std::io::stdin().read_line(&mut line)?;
+            // remove new line char if present and leading and tracing spaces
+            line = line.trim().to_string();
+            match line.to_lowercase().as_str() {
+                "q" | "quit" | "exit" => break,
+                "c" | "continue" => {
+                    self.example_eavesdropped_package(iter);
+                    iter += 1;
+                }
+                _ => continue,
+            }
+        }
+
+        Ok(())
+    }
+
+    fn reinitialize_enc_dec(&mut self) {
+        let mut rng = rand::thread_rng();
+        self.key = rng.gen();
+        self.iv = rng.gen();
+    }
+
+    fn encrypt_all_texts(&mut self) {
+        // clear every CT in CT_vec
+        for ciphertext_vec in &mut self.ciphertexts_vec {
+            ciphertext_vec.clear();
+        }
+        self.ciphertexts_vec.clear();
+        // clear known CT
+        self.known_ciphertext.clear();
+
+        self.known_ciphertext = self.encrypt_text(&self.known_plaintext);
+
+        for text_vec in &self.plaintexts_vec {
+            let encrypted_value = self.encrypt_text(text_vec);
+            self.ciphertexts_vec.push(encrypted_value);
+        }
+    }
+
+    fn encrypt_text(&self, plaintext_vec: &[u8]) -> Vec<u8> {
+        Aes128CbcEnc::new(&self.key.into(), &self.iv.into())
+            .encrypt_padded_vec_mut::<Pkcs7>(plaintext_vec)
+    }
+
+    fn decrypt_text(&self, ciphertext_vec: &[u8]) -> Vec<u8> {
+        Aes128CbcDec::new(&self.key.into(), &self.iv.into())
+            .decrypt_padded_vec_mut::<Pkcs7>(ciphertext_vec)
+            .ok()
+            .unwrap()
+    }
+
+    fn print_known_texts(&self) {
+        let (formatted_blocks_pt, formatted_known_pt) =
+            CBCGame::format_blocks_utf8(&self.known_plaintext, 16);
+        let (formatted_blocks_ct, formatted_known_ct) =
+            CBCGame::format_blocks_hex(&self.known_ciphertext, 16);
+
+        println!("Getting corresponding known CT to leaked PT:");
+        println!("Here's leaked plaintext:");
+        println!("{formatted_blocks_pt}");
+        println!("{formatted_known_pt}\n");
+        println!("Here's corresponding CT to leaked PT:");
+        println!("{formatted_blocks_ct}");
+        println!("{formatted_known_ct}\n\n");
```

Comment:
Let's not use "CT" and "PT" in prints, let's use "ciphertext" and "plaintext" instead.